src/qm.rs

```diff
@@ -2976,16 +1681,19 @@ static qm_tbl_32x32_t: [[[u8; (32 * (32 + 1)) / 2]; 2]; 15] = [
 
 const fn subsampled<const N: usize, const M: usize>(
     src: &[u8; N],
-    sz: usize,
-    step: usize,
+    h: usize,
+    hstep: usize,
+    vstep: usize,
 ) -> [u8; M] {
-    assert!((sz * step) * (sz * step) == N);
-    assert!(sz * sz == M);
+    assert!(N == h * 32);
+    assert!(M * hstep * vstep == h * 32);
     let mut dst = [0; M];
+    let mut i = 0;
 
-    const_for!(y in 0..sz => {
-        const_for!(x in 0..sz => {
-            dst[y * sz + x] = src[y * sz * step * step + x * step];
+    const_for!(y in 0..h/vstep => {
+        const_for!(x in 0..32/hstep => {
+            dst[i] = src[(y * vstep + (vstep - 1) / 2) * 32 + x * hstep + (hstep - 1) / 2];
+            i += 1;
         });
     });
```

Comment:
I think it might be clearer now to write these loops like they are in C.  It'd be `for y in in (0..h).step_by(vstep)` normally, but we can't have `for` loops in `const fn`s, so we need to use a `while` loop, like how `c2rust transpile` outputs things.

Reply:
The `const_for!` macro can actually take a `step_by` parameter, so I used that

Reply:
Oh it can? Cool! Guess I forgot.  Yeah that works even better then.